internal/pkg/template/templates/environment/cf.yml

```diff
@@ -278,6 +284,15 @@ Resources:
     Condition: CreateALB
     Type: AWS::ElasticLoadBalancingV2::LoadBalancer
     Properties:
+      {{- if .PublicHTTPConfig.ELBAccessLogs }}
+      LoadBalancerAttributes:
+        - Key: 'access_logs.s3.enabled'
+          Value: true
+        - Key: 'access_logs.s3.prefix'
+          Value: {{ .PublicHTTPConfig.ELBAccessLogs.BucketPrefix }}
+        - Key: 'access_logs.s3.bucket'
+          Value: {{- if .PublicHTTPConfig.ELBAccessLogs.BucketName }} {{ .PublicHTTPConfig.ELBAccessLogs.BucketName }}{{- else }} !Ref ELBAccessLogsBucket {{- end }}
```

Comment:
`ELBAccessLogsBucket` is created only when we `ShouldCreateBucket`, which returns `true` only when `ELBAccessLogs.BucketName` is not empty. My worry is that we need to go through through a relatively longer chain of logic for us to validate that it works correctly. Do you think we could simplify the logic a little?

For example, do we need `ShouldCreateBucket` - could
```
{{- if .PublicHTTPConfig.ELBAccessLogs.ShouldCreateBucket }}
```
be replaced by
```
{{- if and .PublicHTTPConfig.ELBAccessLogs .PublicHTTPConfig.ELBAccessLogs.BucketName}}
```

Reply:
As you mentioned I had the exact same logic [here](https://github.com/aws/copilot-cli/pull/3849/commits/98f346b8e7496c77e251d6e30b417448e81b210b#diff-e03bdf3cbbc34b11b5dca5b60e3916cbb5fb2b13cb17633661db00ce6c0f2561R59). But Efe was mentioning that we can have [separate method](https://github.com/aws/copilot-cli/pull/3849#discussion_r938325056) as we are checking this same condition twice while creating the bucket and also while importing the `regionalmaps` in the cf.yml file. Let me know what do you think about it.

Reply:
Ditto. I feel `{{- if and .PublicHTTPConfig.ELBAccessLogs .PublicHTTPConfig.ELBAccessLogs.BucketName}}` is probably ok to repeat., since it is quite straight forward.